nautobot/ipam/querysets.py

```diff
@@ -271,6 +272,66 @@ def filter(self, *args, prefix=None, **kwargs):
             kwargs["broadcast"] = last_ip
         return super().filter(*args, **kwargs)
 
+    def _validate_cidr(self, value):
+        """Validate whether ``value`` is a validr IPv4/IPv6 CIDR."""
+        try:
+            cidr = netaddr.IPNetwork(str(value))
+        except netaddr.AddrFormatError:
+            raise ValidationError({"cidr": f"{value} does not appear to be an IPv4 or IPv6 network."})
+        return cidr
+
+    def get_closest_parent(self, cidr, namespace, prefix_length=0):
+        """
+        Return the closest matching parent Prefix for a `cidr` even if it doesn't exist in the database.
+
+        Args:
+            cidr (str): IPv4/IPv6 CIDR string
+            namespace (Namespace): Namespace instance
+            prefix_length (int): Maximum prefix length depth for closest parent lookup
+        """
+        # Validate that it's a real CIDR
+        cidr = self._validate_cidr(cidr)
+        broadcast = cidr.broadcast or cidr.ip
+        ip_version = cidr.version
+
+        try:
+            prefix_length = int(prefix_length)
+        except ValueError:
+            raise ValidationError({"prefix_length": f"Invalid prefix_length: {prefix_length}."})
+
+        # Walk the supernets backwrds from smallest to largest prefix.
+        try:
+            supernets = cidr.supernet(prefixlen=prefix_length)
+        except ValueError as err:
+            raise ValidationError({"prefix_length": str(err)})
+        else:
+            supernets.reverse()
+
+        # Enumerate all unique networks and prefixes
+        networks = {str(s.network) for s in supernets}
+        prefix_lengths = {s.prefixlen for s in supernets}
+        del supernets  # Free the memory because DevOps.
+
+        # Prepare the queryset filter
+        lookup_kwargs = {
+            "network__in": networks,
+            "prefix_length__in": prefix_lengths,
+            "ip_version": ip_version,
+            "broadcast__gte": str(broadcast),
+        }
+        if namespace is not None:
+            lookup_kwargs["namespace"] = namespace
```

Comment:
This is vestigial from initial attempt to make `namespace` optional. It's required, so that should be reflected here.